menuxconfig.py

```diff
@@ -0,0 +1,765 @@
+AUTHOR BE LIABLE FOR ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL
+DAMAGES OR ANY DAMAGES WHATSOEVER RESULTING FROM LOSS OF USE, DATA
+OR PROFITS, WHETHER IN AN ACTION OF CONTRACT, NEGLIGENCE OR OTHER
+TORTIOUS ACTION, ARISING OUT OF OR IN CONNECTION WITH THE USE OR
+PERFORMANCE OF THIS SOFTWARE.
+"""
+
+try:
+    # for Python2
+    import Tkinter as tk
+    import ttk as ttk
+    import ScrolledText as tkst
+    import tkMessageBox as tkmsgbx
+    import tkFileDialog as tkfile
+except ImportError:
+   # for Python3
+    import tkinter as tk
+    import tkinter.ttk as ttk
+    import tkinter.scrolledtext as tkst
+    import tkinter.messagebox as tkmsgbx
+    import tkinter.filedialog as tkfile
+
+from kconfiglib import Kconfig, \
+    Symbol, Choice, MENU, COMMENT, \
+    BOOL, TRISTATE, STRING, INT, HEX, UNKNOWN, \
+    expr_value, \
+    TRI_TO_STR, standard_config_filename
+
+
+def usage(parser, err_str):
+    """ Helper for printing usage
+    """
+    print(err_str + '\n')
+    parser.print_help()
+
+
+def main():
+    """ Main entry point
+    """
+
+    parser = argparse.ArgumentParser(prog=__file__,
+                                     description=information,
+                                     formatter_class=argparse.ArgumentDefaultsHelpFormatter)
+    optional = parser._action_groups.pop()
+    optional.add_argument(
+        '-k', '--kfile', help="Root Kconfig File", action="store",
+        default='Kconfig')
+    optional.add_argument(
+        '-o', '--ofile', help="Config File to use", action="store",
+        default=standard_config_filename())
+    optional.add_argument(
+        '-t', '--title', help="Title to use", action="store",
+        default='Configuration Utility')
+    optional.add_argument(
+        '-H', '--header', help="Header to use in config file", action="store",
+        default=None)
+    optional.add_argument(
+        '-C', '--tree_closed', help="Keep the tree closed by default", action="store_true",
+        default=False)
+
+    parser._action_groups.append(optional)
+    cmd_args, unknown = parser.parse_known_args()
+
+    kfile = cmd_args.kfile
+
+    if len(unknown) == 1:
+        kfile = ''.join(unknown)
+
+    if len(unknown) > 1:
+        usage("Error: Unknown set of optional arguments: " + str(unknown))
+        sys.exit(1)
+
+    cfile = cmd_args.ofile
+    title = cmd_args.title
+    header = cmd_args.header
+    if cmd_args.tree_closed is True:
+        default_open = False
+    else:
+        default_open = True
+
+    kfile = os.path.abspath('.').replace('\\', '/') + "/" + kfile
+    if not os.path.exists(kfile):
+        usage(parser, 'Error: ' + kfile + ": Does not exist")
+        sys.exit(2)
+
+    cfile = os.path.abspath('.').replace('\\', '/') + "/" + cfile
+
+    try:
+        conf = Kconfig(kfile)
+    except Exception as exc:
+        usage(parser, 'Error: Failed to parse ' + kfile)
+        raise (exc)
+        sys.exit(3)
```

Comment:
Redundant parentheses around `exc`, and the `sys.exit(3)` is unreachable.

`sys.exit()` has a cool feature btw: If you pass it (something that can be converted to) a string, it will print it to stderr and exit(1). Maybe you could do this instead, unless the exit status of 3 is super important:

```python
except Exception as exc:
    sys.exit(exc)
```

Exception messages from Kconfiglib point out the location of the error already.